packages/bento-design-system/src/DateField/DateField.tsx

```diff
@@ -25,104 +23,125 @@ type SingleDateFieldProps = {
 type RangeDateFieldProps = {
   type: "range";
   shortcuts?: ShortcutProps<[Date, Date]>[];
-} & FieldProps<[Date | null, Date | null]>;
+} & FieldProps<[Date, Date] | null>;
```

Comment:
this is breaking and I would like to understand the impact: previously `value` and `onChange` could deal with partial ranges (i.e. either start or end date could be null). Now the types imply that we only deal with complete ranges (either the range is fully defined, or it's `null`).

What are the functional implications: what happens if we set the start date and then leave the field? Does it get erased?

Reply:
Yes, in react-aria a range is defined only when both the start and end date are defined. When you start picking a range from the calendar, the start date in the input is not even updated until you pick also the end date. If you write the date manually in the input, the text you type is stored locally and the onChange is triggered only when you complete typing a valid date range.
So it's basically impossible to set only the start date. You have separate states for the selected range (null or complete range) and, eventually, the partial text typed manually in the inputs